### ISSUE TITLE
Update ESLint to Latest Version

### ISSUE THREAD

**Issue by harsha509** [LambdaTest/wdio-lambdatest-service#91]
### Description:

The project is currently using an outdated version of ESLint. To ensure better performance, security, and access to the latest features, we need to update ESLint to the latest version. Additionally, we should verify that all linters are functioning correctly after the upgrade.

## Tasks:

* Upgrade ESLint to the latest version.
* Update any related dependencies (e.g., ESLint plugins, configs).
* Run the linting process and address any issues or errors that arise.
* Ensure that all existing linting rules are maintained and properly configured.
* Verify that the CI/CD pipeline includes linting checks after the upgrade.

## Expected Outcome:

* ESLint is updated to the latest version.
* All linters are functioning correctly with no issues reported.

## Additional Notes:

* Review the ESLint migration guide for any breaking changes that may affect our current setup.

**Comment by theharshdadhich** [comment]
I can work on this issue, there are some dev dependencies missing, just need to install them and update the lint step in CI configuration.

**Comment by Delta456** [comment]
Hi, @theharshdadhich! Thanks for the interest! I have assigned you the issue 😁

**Comment by theharshdadhich** [comment]
Thanks @Delta456 , I'm working on it.

**Comment by theharshdadhich** [comment]
Some testcases are skipping what should I do?
![image](https://github.com/user-attachments/assets/83c4acef-657d-481f-8378-cd20d5a95562)

**Comment by harsha509** [comment]
> Some testcases are skipping what should I do? ![image](https://private-user-images.githubusercontent.com/121938001/373500026-83c4acef-657d-481f-8378-cd20d5a95562.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgwMTU2NjAsIm5iZiI6MTcyODAxNTM2MCwicGF0aCI6Ii8xMjE5MzgwMDEvMzczNTAwMDI2LTgzYzRhY2VmLTY1N2QtNDgxZi04Mzc4LWNkMjBkNWE5NTU2Mi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAwNFQwNDE2MDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lZTFkOGQ1YjNhZDY4MzY2ZDNmYmI3NzYyYTE4Nzc3NDJiNDNjZjI4NzUxMDgxYzdhNWUyYTcxZDcyZmNhNDBiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Owcc7kgfSZ7ZsVdmfUqZEGBqSnIi33LgX5rxLxVH0V8)

HI @theharshdadhich,

You can ignore the skipped tests!
Main objective is to implement `eslint` and add lint checks in CI

**Comment by theharshdadhich** [comment]
okay, thanks

**Comment by theharshdadhich** [comment]
I have updated the issue. See here.
https://github.com/LambdaTest/wdio-lambdatest-service/pull/95

**Comment by nikunj-kohli** [comment]
I would like to contribute here, I request you to assign me

**Comment by navin772** [comment]
@nikunj-kohli I have assigned you this issue, the other issues you have commented on already have a closing PR, see https://github.com/LambdaTest/wdio-lambdatest-service/issues/90#issuecomment-3441127303

**Comment by nikunj-kohli** [comment]
> [@nikunj-kohli](https://github.com/nikunj-kohli) I have assigned you this issue, the other issues you have commented on already have a closing PR, see [#90 (comment)](https://github.com/LambdaTest/wdio-lambdatest-service/issues/90#issuecomment-3441127303)

this repo is eligible for lambda amazon rewards, right?

**Comment by navin772** [comment]
> this repo is eligible for lambda amazon rewards, right?

Yes, the repos mentioned here are eligible for rewards (including the supporting projects) - https://github.com/LambdaTest/lambdatest-hacktoberfest-2025#-projects-you-can-contribute-to
